cmd/server-main.go

```diff
@@ -633,22 +633,41 @@ func serverMain(ctx *cli.Context) {
 		logger.Info(color.RedBold(msg))
 	}
 
-	bootstrapTrace("initializing the server")
-	if err = initServer(GlobalContext, newObject); err != nil {
-		var cerr config.Err
-		// For any config error, we don't need to drop into safe-mode
-		// instead its a user error and should be fixed by user.
-		if errors.As(err, &cerr) {
-			logger.FatalIf(err, "Unable to initialize the server")
+	var buckets []BucketInfo
```

Comment:
you can't modify `buckets` slice inside a go-routine while someone can read from it in the meantime.? 

Whoever needs the list of buckets must also be initialized in the same goroutine @Praveenrajmani

Reply:
It is not a go-routine - just a functional block. 

We wait for the notification subsystem to be done and then unfreeze and proceed with other initializations.

Reply:
what we need is for this to be run in a goroutine but freeze it and let the server start, admin calls are possible to verify what is going on.

That is the advantage of using freeze is that we can start and admin operations can be performed however all S3 operations are blocked.

Reply:
understood